dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/AuthorizationRestRepository.java

```diff
@@ -168,11 +193,69 @@ public AuthorizationRest findOne(Context context, String id) {
             context.switchContextUser(user);
         }
 
+        List<Authorization> authorizations =
+                findAuthorizationsByUUIDList(context, type, uuidList, user, featureNames);
+
+        if (currUser != user) {
+            // restore the real current user
+            context.restoreContextUser();
+        }
+        return converter.toRestPage(authorizations, null, utils.obtainProjection());
+    }
+
+    private List<Authorization> findAuthorizationsByUUIDList(
+        Context context,
+        String type, List<String> uuidList, EPerson user,
+        List<String> featureNames) {
+
+        if (featureNames.isEmpty()) {
+            return new ArrayList<>();
+        }
+
+        List<Authorization> authorizations = new ArrayList<>();
+
+        List<BaseObjectRest> objects = uuidList.stream()
+            .map(uuid -> utils.getBaseObjectRestFromTypeAndUUID(context, type, uuid))
+            .collect(Collectors.toList());
+
+        objects.forEach(object ->
+            featureNames.forEach(featureName -> {
+                try {
+                    authorizations.addAll(authorizationsForObject(context, user, featureName, object));
+                } catch (Exception ex) {
+                    log.error(ex.getMessage(), ex);
```

Comment:
This should likely provide a more useful error message, or it should just say `log.error(ex);`.  There's no reason to pull out the getMessage() here.

Reply:
hi @tdonohue ok, I have added an error message stating the error while retrieving the authorizations. I think having the exception message as part of this message, and then the whole exception thrown in following can better indicate what happened.